app/main.py

```diff
@@ -1,3 +1,63 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, int | float):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, float | int):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Distance | float | int) -> Distance:
```

Comment:
Since you indicate in the type annotation that `other` could be `Distance` class instance as well, either add it's processing in the method or remove from the type annotation. This relates to the `__truediv__` method as well